src/types/ProjectToDo.ts

```diff
@@ -57,7 +57,8 @@ export type DeliverableSearchResultType = {
   name: string;
   position: number;
   status: DeliverableStatusType | null;
-  type: DeliverableTypeType;
+  type: string;
+  'type(raw)': DeliverableTypeType;
```

Comment:
Are these reversed?

Reply:
I think that's correct because `type` should be a `string` since it contains a localized value of `type(raw)`, but perhaps I'm thinking about it incorrectly or mixed things up a bit here...

Reply:
If this is the direct search result, everything is a string. Things can only be cast to `DeliverableTypeType`, `number`, or anything that isn't a `string` explicitly through some sort of hydration. It looks like it is used in `projectToDoAsyncThunk` and `ProjectToDo`, but it is cast to the response from `SearchService.search`. So technically these types are incorrect.

Reply:
In other words, since we are casting the value, and the `SearchResponseElement` type returned from `SearchService.search` allows the cast, we are losing type safety because we are kinda tricking the compiler. This could lead to runtime errors if we attempt to do number things to non-numbers. Also since we aren't using a typeguard, the types that are specific values (like `DeliverableStatusType`, `DeliverableCategoryType`, `DeliverableTypeType`) can actually hold any string here, so a switch statement somewhere else might incorrectly think all cases have been exhausted.

Reply:
Sorry this was probably my mistake to begin with. I misused casting in order to get search results to fit with our API enum types.

Reply:
Ideally a service method is responsible for executing the API call and hydrating the response into the expected / usable type. This way you can use "normal" types around the app and contain the Search API response type

Reply:
Ok thanks for the explanation @nickgraz 🙌  Based on that what changes would you recommend here?

Reply:
I think [all of this](https://github.com/terraware/terraware-web/blob/main/src/redux/features/projectToDo/projectToDoAsyncThunk.ts#L15-L57) should go into a service file somewhere (TodoService?) and we should hydrate the result into the expected type similar to [this](https://github.com/terraware/terraware-web/blob/main/src/services/ParticipantsService.ts#L78).

Reply:
I can get started on refactoring this logic into ToDoService, and you can rebase this PR on top of it. Does that sound okay with you Josh?

Reply:
Ok got it, thanks @nickgraz 🙌  And that sounds great, thanks @tommylau523 🙌

Reply:
@tommylau523 if you refactor this logic into a ToDoService and wire it up, I'm not sure my PR will be needed any longer. Please feel free to take over this ticket.

Reply:
See here!
https://github.com/terraware/terraware-web/pull/2751

I was able to add species on an active deliverable. Would appreciate some help with testing as well.

Reply:
@tommylau523 do you think this PR is needed anymore after your change was merged in?

Reply:
I think this is not needed!

Reply:
Ok so @tommylau523's PR did resolve the issue I was trying to fix in this PR with the types, but it looks like there's also a `404` happening on Staging when I try to add a species to a project from the edit species view.

Your PR cleaned things up a ton, thanks for doing that work @tommylau523 🙌 💯

I'm guessing there's just a path that needs to be updated, but I'll work out the issue with the `404` and open a new PR.

Reply:
Ok so I don't think it's a path issue after all, I'm getting this error message when I try to add the species _Actinidia deliciosa_ to the project New project 2001 in the Graziano Tree Farm organization on Staging:

> "Species Deliverable for project 38 not found"

Reply:
Perhaps there are projects available in the dropdown that shouldn't be available... 🤔 Will poke around a bit more and work out the issue.

Reply:
Ok yeah so I tried again with the “Sampson Road” project and it worked as expected. I think we likely need to filter the list of projects available in the dropdown to a reduced subset.